libraries/firefox_package.rb

```diff
@@ -74,33 +77,72 @@ def munged_platform
       else
         @munged_platform = new_resource.platform
       end
-    end 
+    end
 
-    def explode_tarball(file, dest_path)
+    def explode_tarball(filename, dest_path)
       directory dest_path do
         recursive true
       end
 
       execute 'untar-firefox' do
-        command "tar --strip-components=1 -xjf #{file} -C #{dest_path}"
+        command "tar --strip-components=1 -xjf #{filename} -C #{dest_path}"
         not_if { ::File.exist?(::File.join(dest_path, 'firefox')) }
       end
     end
 
-    def windows_installer(file, version, lang, req_action)
-      windows_package "Mozilla Firefox #{version} (x86 #{lang})" do
-        source file
+    def parse_version(filename)
+      version = /(.[0-9]\.[0-9])(\.[1-9])?/.match(filename)
+      version
+    end
+
+    def windows_long_version(version)
+      if version.nil?
+        version = parse_version(filename)
+        long_version = "#{version}"
+        if esr?(filename)
+          long_version = "#{parse_version(filename)} ESR"
+        end
+      else
+        long_version = version
+      end
+    end
+
+
+    def esr?(filename)
+      version = /esr/.match(filename)
```

Comment:
To be clear, I meant ignoring the match data. All you were doing is checking for the presence. If you don't need the match data object we can skip initializing a new one.